lib/internal/modules/esm/resolve.js

```diff
@@ -641,7 +672,22 @@ function defaultResolve(specifier, { parentURL } = {}, defaultResolveUnused) {
       throw new ERR_INPUT_TYPE_NOT_ALLOWED();
   }
 
-  let url = moduleResolve(specifier, new URL(parentURL));
+  let url;
+  const base = new URL(parentURL);
+  try {
+    url = moduleResolve(specifier, base);
+  } catch (error) {
+    if (base.protocol === 'file:' && error.code === 'ERR_MODULE_NOT_FOUND') {
+      const found = resolveAsCommmonJS(specifier, fileURLToPath(base));
+      if (found) {
+        const lines = error.stack.split('\n');
+        const hint = `Did you mean to import ${found}?`;
+        error.stack = `${lines.shift()}\n${hint}\n${lines.join('\n')}`;
```

Comment:
After rebasing from master overriding just the error message was not changing the output on `stdout`. Is there a better way to do this that I am missing?

Reply:
I was also thinking, would it be better to move the building of the hint directly in the Error builder in https://github.com/nodejs/node/blob/master/lib/internal/errors.js#L1236?